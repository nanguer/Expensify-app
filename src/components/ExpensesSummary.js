import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import 'numeral/locales/es-es';
import selectExpenses from '../selectors/expenses';
import SelectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    numeral.locale('es-es');
    const formattedExpensesTotal = numeral(expensesTotal / 100).format(
        '0,0.00$'
    );
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    return (
        <div>
            Viewing {expenseCount} {expenseWord} totalling{' '}
            {formattedExpensesTotal}
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: SelectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);