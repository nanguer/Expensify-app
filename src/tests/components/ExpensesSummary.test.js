import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render Expenses Summary component with one expense', () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={1} expensesTotal={240} />
    );
    expect(wrapper).toMatchSnapshot();
});

test('Should render Expenses Summary component with multiple expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={10} expensesTotal={500340} />
    );
    expect(wrapper).toMatchSnapshot();
});
