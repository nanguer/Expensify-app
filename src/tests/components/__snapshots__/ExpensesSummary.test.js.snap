// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenses Summary component with multiple expenses 1`] = `
<div>
  Viewing 
  10
   
  expenses
   totalling
   
  5.003,40€
</div>
`;

exports[`Should render Expenses Summary component with one expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling
   
  2,40€
</div>
`;
